### PR TITLE
IA-3793 validate xlsform allow xls cleanup trailing blanks

### PR DESCRIPTION
**Pull request by mestachs**
from feedbacks from first tests : handle xls and xlsx, remove trailing blanks in name and list_name

Related JIRA tickets : IA-3793

## Self proofreading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc

Tell us where the doc can be found (docs folder, wiki, in the code...).

## Changes

from feedbacks from first tests : handle xls and xlsx, remove trailing blanks in name and list_name

## How to test

see https://github.com/BLSQ/iaso/pull/1881

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: handle xls and xlsx, remove trailing blanks in name and list_name before validation

Refs: IA-3793
```

### CODE REVIEW

File: iaso/api/form_versions.py [hunk 1]
```diff
@@ -106,6 +106,8 @@ def validate(self, data: typing.MutableMapping):
             # keep xls_file empty to highlight the input in the UI
             raise serializers.ValidationError({"xls_file": "", "xls_file_validation_errors": validation_errors})
 
+        data["xls_file"].seek(0)
```
Comment (kemar) [inline iaso/api/form_versions.py:109]: Is there any reason why you're doing this here and not in `parse_xls_form()`?

File: iaso/odk/validator.py [hunk 2]
```diff
@@ -85,7 +89,10 @@ def group_by_lambda(collection, field_name_lambda):
 
 
 def validate_xls_form(xls_file):
-    excel_file = pd.ExcelFile(xls_file, engine="openpyxl")
+    file_extension = os.path.splitext(xls_file.name)[1].lower()
+    engine = "xlrd" if file_extension == ".xls" else "openpyxl"
```
Comment (kemar) [inline iaso/odk/validator.py:93]: Why are we using two engines? Is this a speed issue?

**Review by kemar — CHANGES_REQUESTED** [review]
Two questions for you. Maybe they could be answered by adding comments in the code?

File: iaso/api/form_versions.py (hunk 1 above)
Comment (mestachs) [reply iaso/api/form_versions.py:109]: The validator "parse" the xls/xslx and the "parse_xls_rom" reparse it.
I think it only "breaks" when the file is a django "upload" file (like in the viewset)
and not when manipulating "plain" file (that's why I left the fix in the viewset, not sure a normal file have the seek(0))

File: iaso/odk/validator.py (hunk 2 above)
Comment (mestachs) [reply iaso/odk/validator.py:93]: because openpyxl = xlsx only 
and xlrd = xls only

**Review by kemar — APPROVED** [review]
LGTM

**Pull request merged**